### ISSUE TITLE
You are putting compiled code into your crate file

### ISSUE THREAD

**Issue by est31** [emk/rust-cld2#4]
I've downloaded all crates on crates.io and versions 1.0.1 and 0.1.0 of cld2-sys occupy two of the top three places when it comes to size. When I open the archive files, I see a bunch of compiled artifacts inside. Are they included intentionally?

**Comment by emk** [comment]
Unless something has gone seriously wrong, they shouldn't be compiled artifacts. Could you please provide a list of what you're seeing, so that I can compare it?

These crates _do_ contain fairly large data tables used to recognize various languages, which we had to do a fair bit of work to successfully compress enough to upload to crates.io.

**Comment by est31** [comment]
@emk an overview of the sizes of the crate files:

```
$ ls -alh crate-archives/cl/d2/cld2-sys/ | sed 's/  / /g' | cut -d" " -f5,9
[...]
392 cld2-sys-0.0.0.crate
6,5M cld2-sys-0.0.2.crate
6,5M cld2-sys-0.0.3.crate
6,5M cld2-sys-0.0.5.crate
6,5M cld2-sys-0.0.6.crate
6,5M cld2-sys-0.0.7.crate
6,5M cld2-sys-0.0.9.crate
16M cld2-sys-0.1.0.crate
6,5M cld2-sys-1.0.0.crate
16M cld2-sys-1.0.1.crate
````

So there seem to be two affected versions: 0.1.0 and 1.0.1.

Object files inside the 1.0.1 .crate file:

```
$ tar -ztvf crate-archives/cl/d2/cld2-sys/cld2-sys-1.0.1.crate | grep "\.o$"
[...]
-rw-rw-r-- 1000/1000     10328 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/cld2_generated_cjk_compatible.o
-rw-rw-r-- 1000/1000   2125312 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/cld2_generated_deltaocta0122.o
-rw-rw-r-- 1000/1000    272816 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/cld2_generated_distinctocta0122.o
-rw-rw-r-- 1000/1000   9537864 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/cld2_generated_quad0122.o
-rw-rw-r-- 1000/1000    268880 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/cld_generated_cjk_delta_bi_32.o
-rw-rw-r-- 1000/1000    168208 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/cld_generated_cjk_uni_prop_80.o
-rw-rw-r-- 1000/1000     12920 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/cld_generated_score_quad_octa_0122.o
-rw-rw-r-- 1000/1000    125152 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/cldutil.o
-rw-rw-r-- 1000/1000     37304 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/cldutil_shared.o
-rw-rw-r-- 1000/1000     65760 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/compact_lang_det.o
-rw-rw-r-- 1000/1000    156768 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/compact_lang_det_hint_code.o
-rw-rw-r-- 1000/1000    336800 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/compact_lang_det_impl.o
-rw-rw-r-- 1000/1000    152864 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/debug.o
-rw-rw-r-- 1000/1000      6440 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/fixunicodevalue.o
-rw-rw-r-- 1000/1000      6536 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/generated_distinct_bi_0.o
-rw-rw-r-- 1000/1000     15400 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/generated_entities.o
-rw-rw-r-- 1000/1000    142744 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/generated_language.o
-rw-rw-r-- 1000/1000     54736 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/generated_ulscript.o
-rw-rw-r-- 1000/1000    247496 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/getonescriptspan.o
-rw-rw-r-- 1000/1000     63432 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/lang_script.o
-rw-rw-r-- 1000/1000     74944 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/offsetmap.o
-rw-rw-r-- 1000/1000    243880 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/scoreonescriptspan.o
-rw-rw-r-- 1000/1000     47704 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/tote.o
-rw-rw-r-- 1000/1000     93552 2015-11-19 19:22 cld2-sys-1.0.1/cld2/internal/utf8statetable.o
```

The same command applied to the 1.0.0 crate file shows nothing (that one is also 10 MB smaller!), and to the 0.1.0 crate file it also shows a list of .o files.

You can have a look for yourself. The URL scheme one can use to download .crate files is:
```
https://crates.io/api/v1/crates/{cratename}/{version}/download
```
Or alternatively, the backend version (the crates.io url above is a redirect to that):
```
https://d19xqa3lc3clo8.cloudfront.net/crates/{cratename}/{cratename}-{version}.crate
```

URLs of the .crate files:

```
https://d19xqa3lc3clo8.cloudfront.net/crates/cld2-sys/cld2-sys-0.1.0.crate
https://d19xqa3lc3clo8.cloudfront.net/crates/cld2-sys/cld2-sys-1.0.0.crate
https://d19xqa3lc3clo8.cloudfront.net/crates/cld2-sys/cld2-sys-1.0.1.crate
```

Indeed, without the object files the crate files are not small, but with the object files included the crate files are more than two times bigger so it does make a difference :).

Thanks for having a look.

**Comment by emk** [comment]
Ah, thank you! Looks like something went wrong with two of the builds.

Anyway, `cld2` is probably about to be deprecated in favor of `whatlang`, which is a native Rust implementation of the same general idea.

**Comment by est31** [comment]
@emk I've created a PR to fix the issue: #5 

Even if cld2 is deprecated I still think it is better to fix the issue :).

**Comment by emk** [comment]
Thank you for the fix. I've merged it, and soft-deprecated `cld2` in favor of `whatlang`.
